Android/src/org/droidplanner/android/proxy/mission/MissionProxy.java

```diff
@@ -245,9 +246,10 @@ public void addWaypoints(List<LatLong> points) {
         double alt = getLastAltitude();
         List<MissionItem> missionItemsToAdd = new ArrayList<MissionItem>(points.size());
         for (LatLong point : points) {
+            // TODO !BB! Make this method the last used frame
             Waypoint waypoint = new Waypoint();
-            waypoint.setCoordinate(new LatLongAlt(point.getLatitude(), point.getLongitude(),
-                    (float) alt));
+            waypoint.getCoordinate().set(point);
```

Comment:
@billbonney Here and below, `Waypoint#getCoordinate()` is not guaranteed to be non-null, so this changes will result in several npe errors.